File: AkkaRemotingExample/LocalNodeApp/src/main/scala/org/akka/essentials/localnode/LocalNodeApplication.scala
package org.akka.essentials.localnode
import com.typesafe.config.ConfigFactory
import akka.actor.ActorSystem
import akka.actor.Props

object LocalNodeApplication {

  def main(args: Array[String]): Unit = {
    // load the configuration
    val config = ConfigFactory.load().getConfig("LocalSys")
    val system = ActorSystem("LocalNodeApp", config)
    val clientActor = system.actorOf(Props[LocalActor])
    clientActor ! "Hello"
    Thread.sleep(4000)
    system.shutdown()
  }
}

File: AkkaRemotingExample/LocalNodeApp/src/main/java/org/akka/essentials/localnode/LocalNodeApplication.java
package org.akka.essentials.localnode;

import com.typesafe.config.ConfigFactory;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;

/**
 * Hello world!
 * 
 */
public class LocalNodeApplication {
	public static void main(String[] args) throws Exception {
		ActorSystem _system = ActorSystem.create("LocalNodeApp",ConfigFactory
				.load().getConfig("LocalSys"));
		ActorRef localActor = _system.actorOf(new Props(LocalActor.class));
		localActor.tell("Hello");

		Thread.sleep(5000);
		_system.shutdown();
	}
}
